### PR TITLE
Add any_space_10 (for #455)

### PR DESCRIPTION
**Pull request by TeranIvy**
This PR is a simple bug fix for Dynamo0.3 API not having `ANY_SPACE_10` in the list of `VALID_ANY_SPACE_NAMES` (issue #455).

### CODE REVIEW

**Comment by TeranIvy** [comment]
Tests on Travis pass and compilation tests with Intel 17.0.1 on my Desktop also pass. Since @rupertford is so kind to offer to review this change, I am passing the branch to him :-)

File: src/psyclone/dynamo0p3.py [hunk 1]
```diff
@@ -75,10 +75,7 @@
     CONTINUOUS_FUNCTION_SPACES
 
 # Valid any_space metadata (general FS, treated as continuous)
-# TODO (issue #455): add any_space_10
-VALID_ANY_SPACE_NAMES = ["any_space_1", "any_space_2", "any_space_3",
-                         "any_space_4", "any_space_5", "any_space_6",
-                         "any_space_7", "any_space_8", "any_space_9"]
+VALID_ANY_SPACE_NAMES = ["any_space_{0}".format(x+1) for x in range(10)]
```
Comment (rupertford) [inline src/psyclone/dynamo0p3.py]: Much neater. I think the comment above should probably be changed from "treated as continuous" to "could be continuous or discontinuous".

File: doc/user_guide/built_ins.rst [hunk 2]
```diff
@@ -201,7 +209,7 @@ sees fit with no change to the algorithm and kernel layers.  ::
   END MODULE solver_mod_psy
 
 This example is distributed with PSyclone and can be found in
-``<PSYCLONEHOME>/examples/dynamo/eg4``.
+``<PSYCLONEHOME>/examples/dynamo/eg3``.
```
Comment (rupertford) [inline doc/user_guide/built_ins.rst]: Well spotted.

File: doc/user_guide/built_ins.rst [hunk 3]
```diff
@@ -17,7 +17,10 @@ specification of what is required without an implementation. Therefore
 the PSy layer is free to implement these operations in whatever way it
 chooses.
 
-.. note:: In general, PSyclone will need to know the types of the arguments being passed to any Built-ins. The parser obtains this information from an API-specific file that contains the metadata for all Built-in operations supported for that API.
+.. note:: In general, PSyclone will need to know the types of the arguments
```
Comment (rupertford) [inline doc/user_guide/built_ins.rst]: As you've changed this file could you please add copyright info and authorship to the top. The dynamo0p3.rst file already has it so you can use that as a template.

File: src/psyclone/dynamo0p3.py (hunk 1 above)
Comment (TeranIvy) [reply src/psyclone/dynamo0p3.py]: Chamged, thanks for the suggestion.

File: doc/user_guide/built_ins.rst (hunk 2 above)
Comment (TeranIvy) [reply doc/user_guide/built_ins.rst]: Thanks.

File: doc/user_guide/built_ins.rst (hunk 3 above)
Comment (TeranIvy) [reply doc/user_guide/built_ins.rst]: Yes, I realised I should have done that pretty much as I sent the branch to review. Added the copyright and autorship info as requested.

**Comment by TeranIvy** [comment]
Addressed the review comments. Travis and local tests with compilation pass. Back for review with @rupertford.

**Pull request merged**